runtime/memory.cu

```diff
@@ -211,6 +211,105 @@ namespace Hopper {
 // Tensor map:
 // https://docs.nvidia.com/cuda/cuda-driver-api/group__CUDA__TENSOR__MEMORY.html
 
+// TMA Loads:
+
+template <int dim>
+struct CpAsyncBulkTensorTileG2SIndex {
+  const TensorMap* descriptor;
+  Array<int32_t, dim> crds;
+  uint32_t mbarrier;
+};
+
+__device__ inline void cpAsyncBulkTensorTileG2S(
+    const CpAsyncBulkTensorTileG2SIndex<1>& src,
+    uint32_t smem_addr) {
+  uint64_t gmem_int_desc = reinterpret_cast<uint64_t>(src.descriptor);
+  asm volatile(
+      "cp.async.bulk.tensor.1d.shared::cluster.global.mbarrier::complete_tx::bytes"
+      " [%0], [%1, {%3}], [%2];"
```

Comment:
Is there a reason why indicines are not defined in order they are used? (I got confused at the beginning, but this it is correct, only confusing).

Reply:
This segment is copy-pasted from cutlass. I guess the idea for this code style is to push the N-dimensional coordinate to the end of argument list, so that it looks like a parameter pack

```
template<typename ... Ts>
auto f(Ts... x) {
    return (x + ...);
}

int main() {
    std::cout << f(1) << std::endl;
    std::cout << f(1, 2) << std::endl;
    std::cout << f(1, 2, 3) << std::endl;
}
```